homeassistant/components/nibe_heatpump/climate.py

```diff
@@ -0,0 +1,220 @@
+
+        mode = HVACMode.OFF
+        if _get_value(self._coil_use_room_sensor) == "ON":
+            if _get_value(self._coil_cooling_with_room_sensor) == "ON":
+                mode = HVACMode.HEAT_COOL
+            else:
+                mode = HVACMode.HEAT
+        self._attr_hvac_mode = mode
+
+        setpoint_heat = _get_float(self._coil_setpoint_heat)
+        setpoint_cool = _get_float(self._coil_setpoint_cool)
+
+        if mode == HVACMode.HEAT_COOL:
+            self._attr_target_temperature = None
+            self._attr_target_temperature_low = setpoint_heat
+            self._attr_target_temperature_high = setpoint_cool
+        elif mode == HVACMode.HEAT:
+            self._attr_target_temperature = setpoint_heat
+            self._attr_target_temperature_low = None
+            self._attr_target_temperature_high = None
+        else:
+            self._attr_target_temperature = None
+            self._attr_target_temperature_low = None
+            self._attr_target_temperature_high = None
+
+        if prio := _get_value(self._coil_prio):
+            if (
+                _get_value(self._coil_mixing_valve_state)
+                in VALUES_MIXING_VALVE_CLOSED_STATE
+            ):
+                self._attr_hvac_action = HVACAction.IDLE
+            elif prio in VALUES_PRIORITY_HEATING:
+                self._attr_hvac_action = HVACAction.HEATING
+            elif prio in VALUES_PRIORITY_COOLING:
+                self._attr_hvac_action = HVACAction.COOLING
+            else:
+                self._attr_hvac_action = HVACAction.IDLE
+        else:
+            self._attr_hvac_action = None
+
+        self.async_write_ha_state()
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        coordinator = self.coordinator
+        active = self._coil_active_accessory
+
+        if not coordinator.last_update_success:
+            return False
+
+        if not active:
+            return True
+
+        if active_accessory := coordinator.get_coil_value(active):
+            return active_accessory == "ON"
+
+        return False
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set target temperatures."""
+        coordinator = self.coordinator
+        hvac_mode = kwargs.get(ATTR_HVAC_MODE, self._attr_hvac_mode)
+
+        if (temperature := kwargs.get(ATTR_TEMPERATURE)) is not None:
+            if hvac_mode == HVACMode.HEAT:
+                await coordinator.async_write_coil(
+                    self._coil_setpoint_heat, temperature
+                )
+            elif hvac_mode == HVACMode.COOL:
+                await coordinator.async_write_coil(
+                    self._coil_setpoint_cool, temperature
+                )
+            else:
+                raise ValueError(
+                    f"Don't known which temperature to control for hvac mode: {self._attr_hvac_mode}"
+                )
```

Comment:
Maybe change the message to  `"set_temperature" requires "hvac_mode" when passing "temperature" when "hvac_mode" is not already set to "heat" or "cool"`

Reply:
https://github.com/home-assistant/core/pull/83027